pkg/lisafs/message.go

```diff
@@ -1445,30 +1445,32 @@ func (*UnlinkAtResp) String() string {
 }
 
 // RenameAtReq is used to make RenameAt requests. Note that the request takes in
-// the to-be-renamed file's FD instead of oldDir and oldName like renameat(2).
+// the to-be-renamed file's FD instead of oldDir and oldName like renameat2(2).
 type RenameAtReq struct {
 	OldDir  FDID
 	NewDir  FDID
 	OldName SizedString
 	NewName SizedString
+	Flags   primitive.Uint32
```

Comment:
So we can not modify these RPC messages for backward compatibility. You will need to create a new RenameAt2 RPC altogether. For runsc, this works fine because the gofer and the sentry are packaged together.

But internally we may package the gofer separately and they have different release cycles. So you could be running an old sentry and newer gofer and vice versa. Changing the message wire format will break things and lead to unexpected behavior.

In the client side (in fsimpl/gofer/) you will need to check for Server support for an RPC. You will need to do something like:
```
if flags&RENAME_EXCHANGE != 0 {
    if !fs.client.IsSupported(RenameAt2) {
        // The server does not support RenameAt2 RPC.
        return EINVAL
    }
    ...
}
...
if flags&RENAME_EXCHANGE == 0 {
    err = oldParent.controlFDLisa.RenameAt(ctx, oldName, newParent.controlFDLisa.ID(), newName)
} else {
    // make your renameat2 RPC.
}
```

Reply:
New code should fix that